rclcpp_lifecycle/src/state.cpp

```diff
@@ -25,43 +25,61 @@
 namespace rclcpp_lifecycle
 {
 
-State::State()
-: State(lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN, "unknown")
+State::State(rcutils_allocator_t allocator)
+: State(lifecycle_msgs::msg::State::PRIMARY_STATE_UNKNOWN, "unknown", allocator)
 {}
 
-State::State(uint8_t id, const std::string & label)
-: owns_rcl_state_handle_(true)
+State::State(
+  uint8_t id,
+  const std::string & label,
+  rcutils_allocator_t allocator)
+: allocator_(allocator),
+  owns_rcl_state_handle_(true),
+  state_handle_(nullptr)
 {
   if (label.empty()) {
     throw std::runtime_error("Lifecycle State cannot have an empty label.");
   }
 
-  auto state_handle = new rcl_lifecycle_state_t;
+  auto state_handle = reinterpret_cast<rcl_lifecycle_state_t *>(
```

Comment:
A `static_cast` should be used to cast a `void *` to a type specific pointer.

Same below.